src/gui/transferlistfilterswidget.cpp

```diff
@@ -190,17 +192,43 @@ StatusFilterWidget::~StatusFilterWidget()
 
 void StatusFilterWidget::updateTorrentNumbers()
 {
-    const BitTorrent::TorrentStatusReport report = BitTorrent::Session::instance()->torrentStatusReport();
-
-    item(TorrentFilter::All)->setData(Qt::DisplayRole, tr("All (%1)").arg(report.nbActive + report.nbInactive));
-    item(TorrentFilter::Downloading)->setData(Qt::DisplayRole, tr("Downloading (%1)").arg(report.nbDownloading));
-    item(TorrentFilter::Seeding)->setData(Qt::DisplayRole, tr("Seeding (%1)").arg(report.nbSeeding));
-    item(TorrentFilter::Completed)->setData(Qt::DisplayRole, tr("Completed (%1)").arg(report.nbCompleted));
-    item(TorrentFilter::Paused)->setData(Qt::DisplayRole, tr("Paused (%1)").arg(report.nbPaused));
-    item(TorrentFilter::Resumed)->setData(Qt::DisplayRole, tr("Resumed (%1)").arg(report.nbResumed));
-    item(TorrentFilter::Active)->setData(Qt::DisplayRole, tr("Active (%1)").arg(report.nbActive));
-    item(TorrentFilter::Inactive)->setData(Qt::DisplayRole, tr("Inactive (%1)").arg(report.nbInactive));
-    item(TorrentFilter::Errored)->setData(Qt::DisplayRole, tr("Errored (%1)").arg(report.nbErrored));
+    int nbDownloading = 0;
+    int nbSeeding = 0;
+    int nbCompleted = 0;
+    int nbResumed = 0;
+    int nbPaused = 0;
+    int nbActive = 0;
+    int nbInactive = 0;
+    int nbErrored = 0;
+
+    for (const BitTorrent::TorrentHandle *torrent : asConst(BitTorrent::Session::instance()->torrents())) {
+        if (torrent->isDownloading())
+            ++nbDownloading;
+        if (torrent->isUploading())
+            ++nbSeeding;
+        if (torrent->isCompleted())
+            ++nbCompleted;
+        if (torrent->isResumed())
+            ++nbResumed;
+        if (torrent->isPaused())
+            ++nbPaused;
+        if (torrent->isActive())
+            ++nbActive;
+        if (torrent->isInactive())
+            ++nbInactive;
+        if (torrent->isErrored())
+            ++nbErrored;
+    }
+
+    item(TorrentFilter::All)->setData(Qt::DisplayRole, tr("All (%1)").arg(nbActive + nbInactive));
```

Comment:
Why not just `BitTorrent::Session::instance()->torrents().count()`?

Reply:
Sure, done.